### PR TITLE
OCPBUGSM-31584: added columns to Agent CRD

### PR DESCRIPTION
**Pull request by danielerez**
# Assisted Pull Request

## Description

Added the following columns to 'additionalPrinterColumns'
in Agents CRD:
* Role: The role (master/worker) of the Agent.
* Stage: The HostStage of the Agent.
* Hostname: The hostname of the Agent.
* Requested Hostname: The requested hostname for the Agent.

Note:
Adding a complex jsonPath is not supported yet[*],
so can't add columns that are consist of arrays (as IP/MAC).

[*] https://github.com/kubernetes/kubernetes/pull/101205

## List all the issues related to this PR

- [ ] New Feature
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

The following columns should be printed:
* oc -n assisted-installer get agents:
  * NAME,  CLUSTER, APPROVED,  ROLE,  STAGE
* oc -n assisted-installer get agents -o wide:
  * NAME,  CLUSTER, APPROVED,  ROLE,  STAGE,  HOSTNAME, REQUESTED HOSTNAME

## Assignees

/cc @rollandf 
/cc @filanov 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/controller/api/v1beta1/agent_types.go [hunk 1]
```diff
@@ -222,6 +222,9 @@ type DebugInfo struct {
 // +kubebuilder:subresource:status
 // +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".spec.clusterDeploymentName.name",description="The name of the cluster the Agent registered to."
 // +kubebuilder:printcolumn:name="Approved",type="boolean",JSONPath=".spec.approved",description="The `Approve` state of the Agent."
+// +kubebuilder:printcolumn:name="Role",type="string",JSONPath=".status.role",description="The role (master/worker) of the Agent."
+// +kubebuilder:printcolumn:name="Stage",type="string",JSONPath=".status.progress.currentStage",description="The HostStage of the Agent."
+// +kubebuilder:printcolumn:name="Hostname",type="string",JSONPath=".status.inventory.hostname",description="The hostname of the Agent."
```
Comment (rollandf) [inline internal/controller/api/v1beta1/agent_types.go:227]: Maybe we should have both the hostname from the inventory and the hostname from the spec?
`Requested Hostname` maybe?

**Comment by rollandf** [comment]
BTW, we could add also some columns to ACI and InfraEnv.
@filanov WDYT?

File: internal/controller/api/v1beta1/agent_types.go (hunk 1 above)
Comment (danielerez) [reply internal/controller/api/v1beta1/agent_types.go:227]: yeah, we can add it if useful. though we have quite many columns now, so probably should have it in priority, so would be available only using wide output.

**Review by nmagnezi — COMMENTED** [review]
nice!
any chance you can paste the print output here for review?

**Comment by danielerez** [comment]
> nice!
> any change you can paste the print output here for review?

Sure. For example:
```
NAME                                   CLUSTER                                 APPROVED   ROLE     STAGE                   HOSTNAME                                         REQUESTED HOSTNAME
560f9cc6-f85d-449e-8d02-49bc949c30cf   test-infra-cluster-assisted-installer   true       master   Configuring             test-infra-cluster-assisted-installer-master-1
86422489-b843-4764-8fb3-3b5fe2692c79   test-infra-cluster-assisted-installer   true       master   Configuring             test-infra-cluster-assisted-installer-master-2
fbbf9078-f844-4166-82fa-af1c2bf24f98   test-infra-cluster-assisted-installer   true       master   Writing image to disk   test-infra-cluster-assisted-installer-master-0
```

**Review by nmagnezi — APPROVED** [review]
/lgtm

**Pull request merged**